test/framework/easyconfig.py

```diff
@@ -1204,6 +1204,32 @@ def test_dump_extra(self):
 
         dumped_ec = EasyConfig(testec)
 
+    def test_dump_template(self):
+        """ Test EasyConfig's dump() method for files containing templates"""
+        rawtxt = '\n'.join([
+            'easyblock = "EB_toy"',
+            '',
+            'name = "Foo"',
+            'version = "0.0.1"',
+            '',
+            'homepage = "http://foo.com/"',
+            'description = "foo description"',
+            '',
+            "toolchain = {'version': 'dummy', 'name': 'dummy'}",
+            '',
+            "sources = ['foo-0.0.1.tar.gz']",
+        ])
+
+        handle, testec = tempfile.mkstemp(prefix=self.test_prefix, suffix='.eb')
+        os.close(handle)
+
+        ec = EasyConfig(None, rawtxt=rawtxt)
+        ec.dump(testec)
+        ectxt = read_file(testec)
+
+        regex = re.compile(r"sources \= \['SOURCELOWER_TAR_GZ'\]", re.M)
+        self.assertTrue(regex.search(ectxt), "Pattern '%s' found in: %s" % (regex.pattern, ectxt))
```

Comment:
extend this a bit:

* check that `description` is not templated with `%(name)s`
* also check values of other types, i.e. a string (`configopts`?), a dict (`sanity_check_paths`?), with some other templates (`namelower`?)